tools/kubectl-ownerref/main.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	ns = flag.String("ns", "", "namespace")
+)
+
+func main() {
+	var kubeconfig *string
+	if homeDir := homedir.HomeDir(); homeDir != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(homeDir, ".kube", "config"), "kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "kubeconfig file")
+	}
+
+	flag.Parse()
+
+	if err := run(kubeconfig); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+}
+
+func run(kubeconfig *string) error {
+	if len(os.Args) < 3 {
+		fmt.Println("Please complete command: kubectl ownerref [KIND] [NAME]")
+		os.Exit(1)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	discoveryClientSet, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	clientset, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	apiResourceList, err := discoveryClientSet.ServerPreferredResources()
+	if err != nil {
+		return nil
+	}
+
+	for _, apiResource := range apiResourceList {
+		for _, resource := range apiResource.APIResources {
+			group, version := getGroupVersion(apiResource.GroupVersion)
+			res := schema.GroupVersionResource{
+				Group:    group,
+				Version:  version,
+				Resource: resource.Name,
+			}
+			PrintResource(clientset, res, os.Args[1], os.Args[2])
+		}
+	}
+
+	return nil
+}
+
+func PrintResource(c dynamic.Interface, resource schema.GroupVersionResource, kind, name string) error{
+	listOpts := metav1.ListOptions{}
```

Comment:
Another potential optimization: I think it's possible to just get the ObjectMeta, see e.g. https://github.com/kubernetes/kubernetes/pull/77819